plugins/woocommerce/src/Internal/DataStores/StockNotifications/StockNotificationsDataStore.php

```diff
@@ -455,12 +459,35 @@ public function query( array $args ) {
 			$where_values[] = esc_sql( $args['user_email'] );
 		}
 
+		if ( $args['start_date'] ) {
+			$where[]        = 'date_created_gmt >= %s';
+			$where_values[] = esc_sql( $args['start_date'] );
+		}
+
+		if ( $args['end_date'] ) {
+			$where[]        = 'date_created_gmt < %s';
+			$where_values[] = esc_sql( $args['end_date'] );
+		}
+
+		// ORDER BY clauses.
+		$order_by         = '';
+		$order_by_clauses = array();
+
+		if ( $args['order_by'] && is_array( $args['order_by'] ) ) {
+			foreach ( $args['order_by'] as $what => $how ) {
+				$order_by_clauses[] = $table . '.' . esc_sql( strval( $what ) ) . ' ' . esc_sql( strval( $how ) );
+			}
+		}
+
+		$order_by_clauses = empty( $order_by_clauses ) ? array( $table . '.id ASC' ) : $order_by_clauses;
```

Comment:
Let's avoid having a default ORDER here. It's better for the MySQL optimizer.

Reply:
Done!